FWCore/Framework/interface/EDConsumerBase.h

```diff
@@ -288,21 +298,21 @@ namespace edm {
       unsigned int m_startOfComponentName;
     };
 
-    // TODO We would like to be able to access m_esTokenInfo from the
-    // index in the token, but this is currently not possible. One idea
-    // for this is to order the entries in m_esToken so that all the ones
-    // for transition 0 come first, then the ones for for transition 1
-    // and so on for all the transitions. Within a transition, the
-    // entries would be in the same order in m_esTokenInfo and
-    // esItemsToGetFromTransition_. This is something for future
-    // development and might require a change to SoATuple to support
-    // inserts in the middle of the data structure.
     enum { kESLookupInfo, kESResolverIndex };
     edm::SoATuple<ESTokenLookupInfo, ESResolverIndex> m_esTokenInfo;
-    std::array<std::vector<ESResolverIndex>, static_cast<unsigned int>(edm::Transition::NumberOfEventSetupTransitions)>
-        esItemsToGetFromTransition_;
+
+    using ResolverIndexContainer =
+        std::array<std::vector<ESResolverIndex>,
+                   static_cast<unsigned int>(edm::Transition::NumberOfEventSetupTransitions)>;
+    ResolverIndexContainer esItemsToGetFromTransition_;
+
     std::array<std::vector<ESRecordIndex>, static_cast<unsigned int>(edm::Transition::NumberOfEventSetupTransitions)>
         esRecordsToGetFromTransition_;
+
+    // Ordered same as m_esTokenInfo, contents are indexes into esItemsToGetFromTransition_
+    std::vector<std::pair<ResolverIndexContainer::size_type, std::vector<ESResolverIndex>::size_type>>
+        consumesIndexToTransitionAndTokenIndex_;
```

Comment:
Is this only used during the filling of the information for the Service callback? If so, I'm not a fan of it hanging around long term.

Reply:
@Dr15Jones Are you suggesting an added step where I loop over all of them and release the memory after completing the signal that makes PathsAndConsumesOfModules available to the services? Currently, the data member is not used after that.

Note that m_esTokenInfo is also not used after that point as far as I can see. Should I call the destructor on that container also?

Note that consumesIndexToTransitionAndTokenIndex_ connects the elements of m_esTokenInfo and esItemsToGetFromTransition_. It is the new thing that resolves the comment that I deleted (see https://github.com/cms-sw/cmssw/pull/46929#discussion_r1904397777)

For a long time I have been expecting some new code to be added that would use m_esTokenInfo, but it has been a long time and that hasn't happened yet...

Reply:
Is any part of `PathsAndConsumesOfModules` used after the signal it is made available to services?

Reply:
If a member data is only used temporarily, I'd prefer it either not be a member data or releases as much memory as it can after it is no longer going to be used.

Reply:
No, it is not used after the signal.

I have not pushed the newest version of the code yet. If you are curious, it is temporarily here: https://github.com/wddgit/cmssw/tree/printoutEventSetupDependencesRework. In the newest version, PathsAndConsumesOfModules is no longer a data member of EventProcessor. It is now a local variable on the stack in the function EventProcessor::beginJob. Its destructor is called when beginJob finishes.

Reply:
> In the newest version, PathsAndConsumesOfModules is no longer a data member of EventProcessor. It is now a local variable on the stack in the function EventProcessor::beginJob. Its destructor is called when beginJob finishes.

I'd think the destruction of the local `PathsAndConsumesOfModules` at the end of beginJob function scope would be sufficient, and additional `PathsAndConsumesOfModules` member memory release would not be needed.

Reply:
consumesIndexToTransitionAndTokenIndex_ and m_esTokenInfo are data members of EDConsumerBase. They are not in PathsAndConsumesOfModules. They are set or used in updateLookup and the signals making PathsAndConsumesOfModules available. So we would still need to release the memory separately from PathsAndConsumesOfModules.

Reply:
Ah, ok, I missed that. Then I'd agree with Chris it would be nice to release the memory when it is not needed anymore.